internal/client/client_interceptor.go

```diff
@@ -34,17 +36,17 @@ var _ Client = interceptedClient{}
 
 type interceptedClient struct {
 	Client
+	ruRuntimeStatsMap *sync.Map
 }
 
 // NewInterceptedClient creates a Client which can execute interceptor.
-func NewInterceptedClient(client Client) Client {
-	return interceptedClient{client}
+func NewInterceptedClient(client Client, ruRuntimeStatsMap *sync.Map) Client {
+	return interceptedClient{client, ruRuntimeStatsMap}
 }
 
 func (r interceptedClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.Request, timeout time.Duration) (*tikvrpc.Response, error) {
 	// Build the resource control interceptor.
-	resourceGroupName := req.GetResourceGroupName()
-	var finalInterceptor interceptor.RPCInterceptor = buildResourceControlInterceptor(ctx, req, resourceGroupName)
+	var finalInterceptor interceptor.RPCInterceptor = buildResourceControlInterceptor(ctx, req, r.getRURuntimeStats(req.GetStartTs()))
```

Comment:
I wonder `point get` use same ts, if run point get workload and run a `explain analazy` may get wrong result?

Reply:
I checked the TiDB code and found that the point get inside an `EXPLAIN ANALYZE` would not use max uint64 due to the following code:

https://github.com/pingcap/tidb/blob/cc56b21242a2ad82a2f079d7f518606f8705fec4/sessiontxn/isolation/optimistic.go#L132

It will not be regarded as a pure point get since it's an `EXPLAIN` SQL after all. So its `start_ts` will remain unique.